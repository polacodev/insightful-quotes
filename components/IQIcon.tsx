import React from "react";
import Ionicons from "@expo/vector-icons/Ionicons";

interface IQIconProps {
  iconName: "heart-dislike-outline" | "heart-outline" | "star-outline" | "share-social-outline";
  size?: number;
  color?: string;
};

const IQIcon = ({ iconName, size = 24, color = "black" }: IQIconProps) => {
  return (
    <Ionicons name={iconName} size={size} color={color} />
  );
};

export default IQIcon;
