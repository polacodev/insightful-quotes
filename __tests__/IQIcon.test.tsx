import { render } from "@testing-library/react-native";

import IQIcon from "@/components/IQIcon";

jest.mock("expo-font", () => ({
  useFonts: jest.fn(),
  isLoaded: jest.fn(),
  loadAsync: jest.fn(),
}));

describe("<IQIcon />", () => {
  test("IQIcon renders correctly and Snapshot", () => {
    const { toJSON } = render(<IQIcon iconName="star-outline" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
