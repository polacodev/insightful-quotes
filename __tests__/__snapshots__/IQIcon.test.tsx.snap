// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IQIcon /> IQIcon renders correctly and Snapshot 1`] = `<Text />`;
